pypdf/_reader.py

```diff
@@ -349,6 +353,20 @@ def close(self) -> None:
         """Close the underlying file handle"""
         self.stream.close()
 
+    def __enter__(self) -> "PdfReader":
+        """Use PdfReader as context manager"""
+        return self
+
+    def __exit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc: Optional[BaseException],
+        traceback: Optional[TracebackType],
+    ) -> None:
+        """Write data to the fileobj."""
+        if self._we_opened:
+            self.close()
```

Comment:
can you add test to maintain test coverage please

Reply:
Additionally, I would prefer something like `_opened_automatically` instead of `_we_opened` to sound "more generic".